src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -637,8 +637,8 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
 
                     case SyntaxKind.ThisConstructorInitializer:
                     case SyntaxKind.BaseConstructorInitializer:
-                        Debug.Assert(_enclosingBinderOpt == null);
-                        this.binder.BindConstructorInitializer(((ConstructorInitializerSyntax)_containingInvocation).ArgumentList, (MethodSymbol)this.binder.ContainingMember(), diagnostics);
+                        Debug.Assert(_enclosingBinderOpt == null || _enclosingBinderOpt == this.binder);
```

Comment:
>Debug [](start = 24, length = 5)

I'm guessing this has nothing to do with switch statements. Can you please keep PRs to the point in the future?